crates/l3/appchain-core-contract-client/src/interfaces/messaging.rs

```diff
@@ -0,0 +1,88 @@
+use color_eyre::Result;
+use common::invoke_contract;
+use common::LocalWalletSignerMiddleware;
+use starknet_core::types::{FieldElement, InvokeTransactionResult};
+use std::sync::Arc;
+
+pub struct Messaging<'a> {
+    client: Arc<&'a LocalWalletSignerMiddleware>,
```

Comment:
It is not a `client` per say. Just the signer middleware.

It seems to me that the name `client` has been used abusively all over this crate. Don't you think? 
Client is `StarknetCoreContractClient` defined in `clients/client`

Reply:
agreed, have refactored to be signer.